Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java

```diff
@@ -78,20 +84,61 @@ protected void onCreate(Bundle savedInstanceState) {
         ivPlayPause = findViewById(R.id.ivPlayPause);
         ivNext = findViewById(R.id.ivNext);
 
-        songObjectId = getIntent().getStringExtra(EXTRA_SONGOBJECTID);
+        songPosition = getIntent().getIntExtra(EXTRA_CURRENT_SONG_POSITION, 0);
+        sortedSongObjectIds = getIntent().getStringArrayListExtra(EXTRA_SONG_LIST);
         playlistObjectId = getIntent().getStringExtra(EXTRA_PLAYLISTOBJECTID);
 
-        albumUrl = getIntent().getStringExtra(EXTRA_ALBUMICONURL);
-        Glide.with(this).load(albumUrl).into(ivAlbum);
-
-        tvTitle.setText(getIntent().getStringExtra(EXTRA_SONGTITLE));
-        tvArtist.setText(getIntent().getStringExtra(EXTRA_SONGARTIST));
+        setSongView();
 
         ivPlayPause.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                playSong(songObjectId);
-                ivPlayPause.setImageResource(R.drawable.pauseicon);
+
+                // play for the first time
+                if (!isPaused && !isPlay) {
+                    playSong();
+                    ivPlayPause.setImageResource(R.drawable.pauseicon);
+                    isPlay = true;
+
+                    // pause song
+                } else if (!isPaused && isPlay) {
+                    pauseSong();
+                    ivPlayPause.setImageResource(R.drawable.playicon);
+                    isPlay = false;
+                    isPaused = true;
+
+                    // resume song after playing
+                } else if (isPaused && !isPlay) {
+                    resumeSong();
+                    ivPlayPause.setImageResource(R.drawable.pauseicon);
+                    isPaused = false;
+                    isPlay = true;
+                }
+            }
+        });
+
+        ivPrevious.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (songPosition == 0) {
+                    Toast.makeText(SongPlaylistActivity.this, "This is the first song!", Toast.LENGTH_SHORT).show();
+                } else {
+                    songPosition--;
+                    setSongView();
+                }
+            }
+        });
+
+        ivNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast.makeText(SongPlaylistActivity.this, "Clicked!", Toast.LENGTH_SHORT);
```

Comment:
Why do you need this toast?

Reply:
This toast is not necessary anymore. I was using it to debug, but should have removed them when I finished